share/models/fields.py

```diff
@@ -550,3 +555,37 @@ def bulk_related_objects(self):
         # https://github.com/django/django/blob/master/django/db/models/deletion.py#L151
         # Disable django cascading deletes for this field
         raise AttributeError('This is a dirty hack')
+
+
+class EncryptedJSONField(models.BinaryField):
+    """
+    This field transparently encrypts data in the database. It should probably only be used with PG unless
+    the user takes into account the db specific trade-offs with TextFields.
+    """
+    prefix = b'jwe:::'
+
+    def get_db_prep_value(self, input_json, **kwargs):
+        if input_json:
```

Comment:
The return value here should be more deterministic. Returning `input_json` could have some weird side effects.

If `input_json` is an empty dict (`{}`) it will attempt to store that python object in the database. The same goes for empty lists and empty strings. Returning `None` would ensure that strange data can't sneak into the database.